scripts/core/editor3/components/Editor3Component.tsx

```diff
@@ -54,6 +54,19 @@ export function getValidMediaType(event) {
     return VALID_MEDIA_TYPES.find((mediaType) => event.dataTransfer.types.indexOf(mediaType) !== -1);
 }
 
+/**
+ * Check if the files dragged from external folder are of allowed type or not
+ * @param mediaType holds type of data being dragged
+ * @param dataTransfer holds data being dragged
+ */
+export function isAllowedMediaType(mediaType: string, dataTransfer: DataTransfer): boolean {
+    if (mediaType === 'Files') {
+        return Object.values(dataTransfer.files).every(
+            (file: File) => /^image/.test(file.type) || /^audio/.test(file.type) || /^video/.test(file.type));
```

Comment:
Don't use regex where exact comparison can be used against `VALID_MEDIA_TYPES`. Why is it a separate function instead of logic being added to `canDropMedia`?

Reply:
`canDropMedia` actually checks if the dragged item belongs to `VALID_MEDIA_TYPES`, which contains `text/html` or `text/uri-list` as well.  But [here](https://github.com/superdesk/superdesk-client-core/blob/66ed3a43f3cda1027fe16af0e240101d71a18e2b/scripts/core/editor3/components/BaseUnstyledComponent.tsx#L54) we have to allow the media only `audio, image, video or superdesk media`, so created a new method.

Reply:
I don't get it. Your code in `isAllowedMediaType` and the code in `canDropMedia` serve the same purpose - determining if media can be dropped. Why do you want to separate it?